bloom-housing/bloom
# feat: lotteries history log

This PR addresses #4052


## Description

Adds the backend and hooks up the frontend for the history log for lotteries.

Also moves lottery endpoints out of the listing service and into the lottery service.

## How Can This Be Tested/Reviewed?

The following events should be captured in the lottery history log: 

1. Listing closed
2. Lottery ran
3. Lottery re-ran
4. Lottery released
5. Lottery retracted
6. Lottery published

Admins should be able to see all actions but partners cannot see run or re-run. The strings are in the [Figma file](https://www.figma.com/design/PAeCVANDmGdTz2Y53ouErM/Lotteries?node-id=3345-7278&t=9EG0uubDtwoKajvW-0) at each step.

This PR captures the lottery status changes using the activity log interceptor and adds a new endpoint to return a formatted lottery history log. We needed to manipulate some of the data to understand when a lottery was run or retracted.

You can test this on the frontend with an admin user and perform all the actions, and then log-in as a partner user once the lottery is released and ensure you can only see the events you are supposed to see.

The re-run button is not currently hooked up, so I was unable to test that flow through the UI but it is covered in tests.

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
